### PR TITLE
fix: [CDS-75505]: Exclude some paths in diff algorithm

### PR DESCRIPTION
**Pull request by srdjan-harness**
### Summary

In the pipeline (when we use pipeline/stage template) for some entities like service when we chose service we are getting additional fields that are set as runtime inputs e.g. tags.
In this case diff-algorithm reporting that pipeline inputs and template inputs are different (in other words this means that template or pipeline(yaml) has been changed).

This pr exclude some paths (entities) for this and other edge cases.

https://harness.atlassian.net/browse/CDS-75505

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by srdjan-harness** [comment]
retrigger titlecheck

**Comment by srdjan-harness** [comment]
retrigger jest

**Comment by srdjan-harness** [comment]
retrigger jest

**Comment by srdjan-harness** [comment]
retrigger jest

**Comment by srdjan-harness** [comment]
retrigger coverage

**Comment by srdjan-harness** [comment]
retrigger coverage

**Comment by srdjan-harness** [comment]
retrigger jest

**Review by PratyushGar — APPROVED** [review]

File: src/modules/70-pipeline/utils/pipelineReconcileUtils.ts [hunk 1]
```diff
@@ -75,7 +87,16 @@ function hasDiffTemplateInputsAndStageOrPipeline<T extends { [key: string]: any
...
+    const fullPath = prefix ? `${prefix}${path}` : path
+    return `${get(obj, fullPath)}`.startsWith('<+input>') && !excludePath(fullPath)
+  })
+}
+
+function excludePath(path: string): boolean {
+  return PATHS_TO_EXCLUDE.some(pathToExclude => {
+    if (path.includes(pathToExclude)) {
+      return true
+    }
```
Comment (rameshx) [inline src/modules/70-pipeline/utils/pipelineReconcileUtils.ts:99]: Nit: `path.includes(pathToExclude)` can be directly returned inside the callback passed to `some`.

File: src/modules/70-pipeline/utils/pipelineReconcileUtils.ts [hunk 2]
```diff
@@ -42,29 +54,29 @@ function hasDiffTemplateInputsAndStageOrPipeline<T extends { [key: string]: any
...
 
   let hasDifference = false
   // NOTE: When we change from runtime input to fixed value, property will be deleted. This will cause a difference in pipeline and template.
   // To prevent this, if yaml is changed (isUpdated) we are not checking if there are more fileds in template than in pipeline.
   // This means that in some edge cases algorithm will not work as expected.
   if (!isUpdated) {
-    templatePaths_WithoutIndex_RuntimeOnly.forEach((tlPath: string) => {
-      if (pipelinePaths_WithoutIndex.indexOf(tlPath) === -1) {
+    templatePaths_RuntimeOnly.forEach((tlPath: string) => {
+      if (pipelinePaths.indexOf(tlPath) === -1) {
```
Comment (rameshx) [inline src/modules/70-pipeline/utils/pipelineReconcileUtils.ts:71]: Nit: Something like `!includes` can be used instead of `index === -1`

**Review by rameshx — APPROVED** [review]

File: src/modules/70-pipeline/utils/pipelineReconcileUtils.ts (hunk 1 above)
Comment (srdjan-harness) [reply src/modules/70-pipeline/utils/pipelineReconcileUtils.ts:99]: I've created one more ticket related to this algorithm, will update when i work on it.
